target/linux/mediatek/dts/mt7981b-nradio-wt9109-second-system.dts

```diff
@@ -0,0 +1,270 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+/dts-v1/;
+
+#include "mt7981.dtsi"
+
+/ {
+	model = "NRadio C8-668GL (Second System)";

```

Comment:
Why it’s called 'Second System'?

Reply:
> Why it’s called 'Second System'?

After flashing the specific official firmware, the device creates several partitions: kernel, kernel_2nd, rootfs, and rootfs_2nd. Among these, the 2nd partition can be used for running the OpenWrt system. The official system supports flashing and switching to the second system via the WebUI.

Reply:
Couldn't the first system run OpenWrt? I don't think you should specify a second system to run, but let the user choose.

Reply:
> Couldn't the first system run OpenWrt? I don't think you should specify a second system to run, but let the user choose.

The first system can run OpenWrt by modifying the DTS and related shell scripts. The second system is specified for the following reasons:

1. This device originally used a single-system setup but was upgraded to a dual-system configuration in later firmware updates.  
2. The official system is stable and supports remote management via the app.  
3. The second system provides a fallback if configuration errors occur.  
4. The first system has significantly less storage (242 MB vs. 6.6 GB).  

If the user wants OpenWrt as the only system, they need to flash the earliest firmware version and modify the DTS and scripts to remove the second system. For dual-system mode with OpenWrt on both systems, similar DTS and script modifications are required.

Reply:
In any case the model name should not include 'Second System'.
Note at bootargs: If the official system version is lower than xxx, then root=xxx.

Reply:
> In any case the model name should not include 'Second System'. Note at bootargs: If the official system version is lower than xxx, then root=xxx.

Good idea! I will add a comment to the `bootargs` section, like this:

```dts
chosen {
	stdout-path = "serial0:115200n1";

	/* If official system version < 1.9.2.n6.c3, use root=PARTLABEL=rootfs */
	bootargs = "console=ttyS0,115200n1 root=PARTLABEL=rootfs_2nd rootwait";
};
```

Additionally, the relevant names will be updated as follows:

```txt
DTS filename: nradio-c8-668gl
DTS model: NRadio C8-668GL
DTS compatible: nradio,c8-668gl
Image definition: nradio_c8-668gl
```